CHANGELOG.md

```diff
@@ -1,5 +1,103 @@
 # [Versions](https://mui.com/versions/)
 
+## 5.14.4
+
+<!-- generated comparing v5.14.3..master -->
+
+_Aug 8, 2023_
+
+A big thanks to the 18 contributors who made this release possible. Here are some highlights ✨:
+
+- 🎉 Added [Number input](https://mui.com/base-ui/react-number-input) component & [useNumberInput](https://mui.com/base-ui/react-number-input#hook) hook in [Base UI](https://mui.com/base-ui/getting-started/) @mj12albert
```

Comment:
I fixed a few issues on this in https://github.com/mui/material-ui/commit/d42063681d7793d55451e1caacf85c1a5a0f61e2.